src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/ResolveDataLocks/AcceptDataLocksRequestChangesCommandHandler.cs

```diff
@@ -60,11 +60,11 @@ protected override async Task Handle(AcceptDataLocksRequestChangesCommand reques
                 var dataLockWithUpdatedTraining = dataLocksToBeAccepted.FirstOrDefault(m => m.IlrTrainingCourseCode != apprenticeship.CourseCode);
                 if (dataLockWithUpdatedTraining != null)
                 {
-                    var training = await _trainingProgrammeLookup.GetTrainingProgramme(dataLockWithUpdatedTraining.IlrTrainingCourseCode);
+                    var training = await _trainingProgrammeLookup.GetCalculatedTrainingProgrammeVersion(dataLockWithUpdatedTraining.IlrTrainingCourseCode, dataLockWithUpdatedTraining.IlrEffectiveFromDate.GetValueOrDefault());
 
                     _logger.LogInformation($"Updating course for apprenticeship {apprenticeship.Id} from training code {apprenticeship.CourseCode} to {dataLockWithUpdatedTraining.IlrTrainingCourseCode}");
 
-                    apprenticeship.UpdateCourse(Party.Employer, dataLockWithUpdatedTraining.IlrTrainingCourseCode, training.Name, training.ProgrammeType, request.UserInfo);
+                    apprenticeship.UpdateCourse(Party.Employer, dataLockWithUpdatedTraining.IlrTrainingCourseCode, training.Name, training.ProgrammeType, request.UserInfo, training.StandardUId);
```

Comment:
Alan suggested we consider the case where it's a standard course in commitments and they try to change it to a Framework code. Should this be allowed? If so, StandardUID will definitely need to be null

Reply:
Thanks Paul, pretty sure from tracing this through that the training object will be null if there are no standard versions associated based on course code and start date. That effectivity would do the UpdateCourse method with a null in place of StandardUId

Reply:
What do we do if the ILR is changing it to a Framework code. I know this is probably a BA question, but do we allow it?

Reply:
Not sure the answer here, not sure what the format of a 'framework code' looks like... I have wrapped up the updateCourse method based on training not being null though.

Reply:
@ramnatarajanuk-dfe Raised this portion here in ticket notes

Reply:
I added back in the old get training programme call when the Standard one returns null. This should ensure the framework codes work as in the original flow